iceoryx_posh/test/moduletests/test_popo_rpc_header.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/popo/rpc_header.hpp"
+
+#include "iceoryx_posh/testing/mocks/chunk_mock.hpp"
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::popo;
+
+class RpcBaseHeaderAccess : public RpcBaseHeader
+{
+  public:
+    using RpcBaseHeader::m_clientQueueUniquePortId;
+    using RpcBaseHeader::m_lastKnownClientQueueIndex;
+};
+
+class RpcBaseHeader_test : public Test
+{
+  public:
+    void SetUp() override
+    {
+    }
+
+    void TearDown() override
+    {
+    }
+
+    ChunkMock<bool, RpcBaseHeader> chunk;
+    RpcBaseHeader* sut{new (chunk.userHeader())
+                           RpcBaseHeader(iox::UniquePortId(), 73, 37, RpcBaseHeader::RPC_HEADER_VERSION)};
+};
+
+void checkRpcBaseHeader(const RpcBaseHeaderAccess* sut,
+                        const iox::UniquePortId& clientQueueUniquePortId,
+                        const uint32_t lastKnownClientQueueIndex,
+                        const int64_t sequenceId,
+                        const uint8_t rpcHeaderVersion)
+{
+    EXPECT_THAT(sut->getRpcHeaderVersion(), rpcHeaderVersion);
+    EXPECT_THAT(sut->m_clientQueueUniquePortId, Eq(clientQueueUniquePortId));
+    EXPECT_THAT(sut->m_lastKnownClientQueueIndex, lastKnownClientQueueIndex);
+    EXPECT_THAT(sut->getSequenceId(), Eq(sequenceId));
+}
+
+TEST_F(RpcBaseHeader_test, ConstructorWorks)
+{
+    const iox::UniquePortId clientQueueUniquePortId;
+    constexpr uint32_t LAST_KNOWN_CLIENT_QUEUE_INDEX{13};
+    constexpr int64_t SEQUENCE_ID{42};
+    constexpr uint8_t RPC_HEADER_VERSION{222};
+
+    ChunkMock<bool, RpcBaseHeader> chunk;
```

Comment:
This line can be removed since `chunk` is already declared in l. 45

Reply:
yes, but that already holds a `RpcBaseHeader` which would be overridden if used here